kratos/utilities/shifted_boundary_meshless_interface_utility.cpp

```diff
@@ -0,0 +1,1124 @@
+                    // The nodes will be flagged if required (MLS basis) when creating the cloud
+                    auto p_neigh_elem = r_neigh_elems(i_face).get();
+                    if (p_neigh_elem != nullptr) {
+                        if (p_neigh_elem->Is(ACTIVE)) {
+                            p_neigh_elem->Set(INTERFACE, true);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bool ShiftedBoundaryMeshlessInterfaceUtility::IsSplit(const GeometryType& rGeometry)
+    {
+        std::size_t n_neg = 0;
+        std::size_t n_pos = 0;
+        for (const auto& r_node : rGeometry) {
+            if (r_node.FastGetSolutionStepValue(DISTANCE) < 0.0) {
+                n_neg++;
+            } else {
+                n_pos++;
+            }
+        }
+        return (n_pos != 0 && n_neg != 0);
+    }
+
+    bool ShiftedBoundaryMeshlessInterfaceUtility::IsNegative(const GeometryType& rGeometry)
+    {
+        std::size_t n_neg = 0;
+        for (const auto& r_node : rGeometry) {
+            if (r_node.FastGetSolutionStepValue(DISTANCE) < 0.0) {
+                n_neg++;
+            }
+        }
+        return (n_neg == rGeometry.PointsNumber());
+    }
+
+    void ShiftedBoundaryMeshlessInterfaceUtility::SetNodalDistancesVector(
+        const GeometryType& rGeometry,
+        Vector& rNodalDistances)
+    {
+        const std::size_t n_nodes = rGeometry.PointsNumber();
+        if (rNodalDistances.size() != n_nodes) {
+            rNodalDistances.resize(n_nodes);
+        }
+        std::size_t i = 0;
+        for (const auto& r_node : rGeometry) {
+            rNodalDistances[i++] = r_node.FastGetSolutionStepValue(DISTANCE);
+        }
+    }
+
+    ShiftedBoundaryMeshlessInterfaceUtility::ModifiedShapeFunctionsFactoryType ShiftedBoundaryMeshlessInterfaceUtility::GetStandardModifiedShapeFunctionsFactory(const GeometryType& rGeometry)
+    {
+        switch (rGeometry.GetGeometryType()) {
+            case GeometryData::KratosGeometryType::Kratos_Triangle2D3:
+                return [](const GeometryType::Pointer pGeometry, const Vector& rNodalDistances)->ModifiedShapeFunctions::UniquePointer{
+                    return Kratos::make_unique<Triangle2D3ModifiedShapeFunctions>(pGeometry, rNodalDistances);};
+            case GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4:
+                return [](const GeometryType::Pointer pGeometry, const Vector& rNodalDistances)->ModifiedShapeFunctions::UniquePointer{
+                    return Kratos::make_unique<Tetrahedra3D4ModifiedShapeFunctions>(pGeometry, rNodalDistances);};
+            default:
+                KRATOS_ERROR << "Asking for a non-implemented modified shape functions geometry.";
+        }
+    }
+
+    ShiftedBoundaryMeshlessInterfaceUtility::MLSShapeFunctionsAndGradientsFunctionType ShiftedBoundaryMeshlessInterfaceUtility::GetMLSShapeFunctionsAndGradientsFunction()

```

Comment:
```suggestion
    auto ShiftedBoundaryMeshlessInterfaceUtility::GetMLSShapeFunctionsAndGradientsFunction() const

```

Reply:
It seems that `auto` return type cannot be used in this case (some lambda-related stuff). OK about the `conts`.